.pre-commit-config.yaml

```diff
@@ -8,105 +8,24 @@ repos:
-                aiida/backends/sqlalchemy/migrations/versions/162b99bca4a2_drop_dbcalcstate.py|
-                aiida/backends/sqlalchemy/models/computer.py|
-                aiida/backends/sqlalchemy/models/settings.py|
-                aiida/backends/sqlalchemy/models/node.py|
-                aiida/backends/utils.py|
-                aiida/common/datastructures.py|
-                aiida/engine/daemon/execmanager.py|
-                aiida/engine/processes/calcjobs/tasks.py|
-                aiida/orm/querybuilder.py|
-                aiida/orm/nodes/data/array/bands.py|
-                aiida/orm/nodes/data/array/projection.py|
-                aiida/orm/nodes/data/array/xy.py|
-                aiida/orm/nodes/data/code.py|
-                aiida/orm/nodes/data/orbital.py|
-                aiida/orm/nodes/data/remote.py|
-                aiida/orm/nodes/data/structure.py|
-                aiida/orm/utils/remote.py|
-                aiida/parsers/plugins/arithmetic/add.py|
-                aiida/parsers/plugins/templatereplacer/doubler.py|
-                aiida/parsers/plugins/templatereplacer/__init__.py|
-                aiida/plugins/entry.py|
-                aiida/plugins/info.py|
-                aiida/plugins/registry.py|
-                aiida/tools/data/array/kpoints/legacy.py|
-                aiida/tools/data/array/kpoints/seekpath.py|
-                aiida/tools/data/__init__.py|
-                aiida/tools/dbexporters/__init__.py|
-                aiida/tools/dbimporters/baseclasses.py|
-                aiida/tools/dbimporters/__init__.py|
-                aiida/tools/dbimporters/plugins/cod.py|
-                aiida/tools/dbimporters/plugins/icsd.py|
-                aiida/tools/dbimporters/plugins/__init__.py|
-                aiida/tools/dbimporters/plugins/mpds.py|
-                aiida/tools/dbimporters/plugins/mpod.py|
-                aiida/tools/dbimporters/plugins/nninc.py|
-                aiida/tools/dbimporters/plugins/oqmd.py|
-                aiida/tools/dbimporters/plugins/pcod.py|
                 docs/.*|
-                examples/.*|
-                tests/engine/test_work_chain.py|
-                tests/schedulers/test_direct.py|
-                tests/schedulers/test_lsf.py|
-                tests/schedulers/test_pbspro.py|
-                tests/schedulers/test_sge.py|
-                tests/schedulers/test_torque.py|
-                tests/sphinxext/workchain_source/conf.py|
-                tests/sphinxext/workchain_source_broken/conf.py|
-                tests/transports/test_all_plugins.py|
-                tests/transports/test_local.py|
-                tests/transports/test_ssh.py|
-                tests/test_dataclasses.py|
             )$
+        args: ['-i']
 
--   repo: https://github.com/pre-commit/mirrors-yapf
-    rev: v0.30.0
+-   repo: https://github.com/PyCQA/pylint
+    rev: pylint-2.5.2
```

Comment:
were we going to move this into the local repo, as suggested by @ltalirz ?

Reply:
But then we are back to the situation where we have to pin an exact version which will collide with any other projects you might have in the same virtual env. Is there no way to address this? I am not sure if this current solution actually allowed for this or if it had the same limitation and I just never noticed the collision

Reply:
I think we still are in this situation - the only difference being that now we create an additional virtual environment that is never used

Reply:
but as we discussed, I'm under the impression that, because of `language: system`, it is already using the virtual env version of the package **not** the one that is installed in the pre-commit env

Reply:
to use the one in the pre-commit env, you would have to either ignore errors regarding dependency installs, or also install them dependencies into the pre-commit environment, using https://pre-commit.com/#config-additional_dependencies

Reply:
@chrisjsewell not sure whether you replied to my comment or seb's - anyhow I did mean we create an additional *pre-commit* environment that is never used, i.e. we are saying the same thing.

As for your following comment, I don't think either of those are viable solutions, i.e. we can just move the hook back under `local`

Reply:
Moved it back to `local`

Reply:
> not sure whether you replied to my comment or seb's

seb's

> I don't think either of those are viable solutions

yep agree, just outlining the possible option 👍